langchain-core/src/language_models/chat_models.ts

```diff
@@ -306,48 +309,88 @@ export abstract class BaseChatModel<
       undefined,
       handledOptions.runName
     );
-    // generate results
-    const results = await Promise.allSettled(
-      baseMessages.map((messageList, i) =>
-        this._generate(
-          messageList,
-          { ...parsedOptions, promptIndex: i },
-          runManagers?.[i]
-        )
-      )
-    );
-    // handle results
-    const generations: ChatGeneration[][] = [];
+    let generations: ChatGeneration[][] = [];
     const llmOutputs: LLMResult["llmOutput"][] = [];
-    await Promise.all(
-      results.map(async (pResult, i) => {
-        if (pResult.status === "fulfilled") {
-          const result = pResult.value;
-          for (const generation of result.generations) {
-            generation.message.response_metadata = {
-              ...generation.generationInfo,
-              ...generation.message.response_metadata,
-            };
-          }
-          if (result.generations.length === 1) {
-            result.generations[0].message.response_metadata = {
-              ...result.llmOutput,
-              ...result.generations[0].message.response_metadata,
-            };
+    // Even if stream is not explicitly called, check if model is implicitly
+    // called from streamEvents() or streamLog() to get all streamed events.
+    // Bail out if _streamResponseChunks not overridden
+    const hasStreamingHandler = !!runManagers?.[0].handlers.find((handler) => {
+      return isStreamEventsHandler(handler) || isLogStreamHandler(handler);
+    });
+    if (
+      hasStreamingHandler &&
+      baseMessages.length === 1 &&
+      this._streamResponseChunks !==
```

Comment:
we don't do anything dumb like bind the method in constructor right?

Reply:
I don't think so, no. We're already relying on this check in another place